armi/reactor/components/__init__.py

```diff
@@ -343,28 +350,40 @@ def _deriveVolumeAndArea(self):
                     f"More than one ``DerivedShape`` component in {self.parent} is not allowed."
                 )
 
-            siblingArea += sibling.getArea()
             siblingVolume += sibling.getVolume()
+            try:
+                if siblingArea is not None:
+                    siblingArea += sibling.getArea()
+            except:
+                siblingArea = None
 
-        remainingArea = self.parent.getMaxArea() - siblingArea
         remainingVolume = self.parent.getMaxVolume() - siblingVolume
+        if siblingArea:
+            remainingArea = self.parent.getMaxArea() - siblingArea
 
-        # Check for negative area
-        if remainingArea < 0:
+        # Check for negative
+        if remainingVolume < 0:
```

Comment:
This negative volume check isn't covered by unit tests.

Thanks!

Reply:
Added in https://github.com/terrapower/armi/pull/525/commits/2ef113d8cf93b5c7f03704048d70bef94fce0f31